File: interpreter-for-cpp/src/Runtime/Value/ValueObject.cpp
#include "ValueObject.h"
#include "../Space.h"

using namespace peak::interpreter;

ValueObject::ValueObject(std::shared_ptr<Space> space)
	: _space(space) {
}

std::shared_ptr<Value> ValueObject::Clone() const {
	return std::shared_ptr<Value>(new ValueObject(_space->CopySpace()));
}

std::shared_ptr<Space> ValueObject::GetSpace() const {
	return _space;
}

std::string ValueObject::ToString() const {
	return "object";
}

File: interpreter-for-cs/src/Runtime/Value/ValueObject.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {

	public class ValueObject : TypeValue<ValueObject> {
		private Space _space = null;
		public Space space { get => _space; }
		public ValueObject(Space space) {
			_space = space;
		}
		public override string ToString() {
			return "object";
		}
		public override Value Clone() {
			return new ValueObject(_space.CopySpace());
		}
	};
} // namespace peak.interpreter